src/components/AlbulList.js

```diff
@@ -0,0 +1,154 @@
+import React, { useState } from 'react';
+import axios from 'axios';
+import IconButton from '@mui/material/IconButton';
+import CloseIcon from '@mui/icons-material/Close';
+import Snackbar from '@mui/material/Snackbar';
+import {
+  Container,
+  TextField,
+  Button,
+  FormControl,
+  InputLabel,
+  Select,
+  MenuItem,
+  Typography,
+  Box,
+  Grid,
+} from '@mui/material';
+
+const AlbumsList = () => {
+  const [albums, setAlbums] = useState([]); // for albums
+  const [limit, setLimit] = useState(1); // for pagination
+  const [searchType, setSearchType] = useState('albumName'); //default value for select input
+  const [searchTerm, setSearchTerm] = useState(''); // for search input filed value
+  const [message, setMessage] = useState(''); // for not found message
+  const [hasError, setHasError] = useState(false); //for error during fetching from API
+
+  //make an API call with search values to backend and return the result
+  const fetchAlbums = async (searchType, searchTerm, limit) => {
+    try {
+      const response = await axios.get(
+        'http://localhost:8000/api/v1/albums/filter',
+        {
+          params: {
+            limit,
+            [searchType]: searchTerm,
+          },
+        }
+      );
+      const searchResult = response.data.albums;
+      setAlbums(searchResult);
+      if (searchResult.length === 0) {
+        setMessage('No result found');
+      }
+    } catch (error) {
+      console.error('Error fetching albums:', error);
+      setHasError(true);
+    }
+  };
+
+  //call the function to make API request for search input value
+  const handleSearch = () => {
+    setLimit(1);
+    fetchAlbums(searchType, searchTerm, 1);
+  };
+
+  //clear search input and make an empty API call
+  const handleClear = () => {
+    setSearchType('albumName');
+    setSearchTerm('');
+    setLimit(1);
+    setMessage('');
+    fetchAlbums('albumName', '', 1);
+  };
+
+  // snackbar start
+  const [open, setOpen] = React.useState(true);
+
+  const handleClose = (event, reason) => {
+    if (reason === 'clickaway') {
+      return;
+    }
+    setOpen(false);
+  };
+
+  const action = (
+    <React.Fragment>
+      <IconButton
+        size="small"
+        aria-label="close"
+        color="inherit"
+        onClick={handleClose}
+      >
+        <CloseIcon fontSize="small" />
+      </IconButton>
+    </React.Fragment>
+  );
+
+  //end of snackbar
+
+  return (
+    <Container>
+      {/* display snackbar if any error happened during API fetch */}
+      {hasError && (
+        <Snackbar
+          open={open}
+          autoHideDuration={6000}
+          onClose={handleClose}
+          message={errorMessage}
```

Comment:
I meant to also share this link to explain a bit about why to use state instead of a plain variable in this case. Basically anytime you're wanting something in the UI to change as a result of changing a variable then it should probably be state or at least derived from state. Otherwise, plain variables are fine. https://stackoverflow.com/questions/58252454/react-hooks-using-usestate-vs-just-variables#:~:text=That%20means%2C%20when%20the%20value,useState%20than%20normal%20local%20value.&text=Local%20vaiables%20value%20doesn't,the%20variable%20value%20between%20renders.